internal/cmd/commands/authmethodscmd/ldap_funcs.go

```diff
@@ -0,0 +1,440 @@
+	useTokenGroupsFlagName       = "use-token-groups"
+	accountAttributeMaps         = "account-attribute-map"
+)
+
+func extraLdapActionsFlagsMapFuncImpl() map[string][]string {
+	flags := map[string][]string{
+		"create": {
+			urlsFlagName,
+			insecureTlsFlagName,
+			discoverDnFlagName,
+			anonGroupSearchFlagName,
+			upnDomainFlagName,
+			startTlsFlagName,
+			userDnFlagName,
+			userAttrFlagName,
+			userFilterFlagName,
+			enableGroupsFlagName,
+			groupDnFlagName,
+			groupAttrFlagName,
+			groupFilterFlagName,
+			certificatesFlagName,
+			clientCertificateFlagName,
+			clientCertificateKeyFlagName,
+			bindDnFlagName,
+			bindPasswordFlagName,
+			useTokenGroupsFlagName,
+			accountAttributeMaps,
+		},
+	}
+	flags["update"] = flags["create"]
+	return flags
+}
+
+func extraLdapFlagsFuncImpl(c *LdapCommand, set *base.FlagSets, _ *base.FlagSet) {
+	f := set.NewFlagSet("LDAP Auth Method Options")
+
+	for _, name := range flagsLdapMap[c.Func] {
+		switch name {
+		case urlsFlagName:
+			f.StringSliceVar(&base.StringSliceVar{
+				Name:   urlsFlagName,
+				Target: &c.flagUrls,
+				Usage:  "The LDAP URLs that specify LDAP servers to connect to (required).  May be specified multiple times.",
+			})
+		case startTlsFlagName:
+			f.BoolVar(&base.BoolVar{
+				Name:   startTlsFlagName,
+				Target: &c.flagStartTls,
+				Usage:  "Issue StartTLS command after connecting (optional).",
+			})
+		case insecureTlsFlagName:
+			f.BoolVar(&base.BoolVar{
+				Name:   insecureTlsFlagName,
+				Target: &c.flagInsecureTls,
+				Usage:  "Skip the LDAP server SSL certificate validation (optional) - insecure and use with caution.",
+			})
+		case discoverDnFlagName:
+			f.BoolVar(&base.BoolVar{
+				Name:   discoverDnFlagName,
+				Target: &c.flagDiscoverDn,
+				Usage:  "Use anon bind to discover the bind DN of a user (optional).",
+			})
+		case anonGroupSearchFlagName:
+			f.BoolVar(&base.BoolVar{
+				Name:   anonGroupSearchFlagName,
+				Target: &c.flagAnonGroupSearch,
+				Usage:  "Use anon bind when performing LDAP group searches (optional).",
+			})
+		case upnDomainFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   upnDomainFlagName,
+				Target: &c.flagUpnDomain,
+				Usage:  "The userPrincipalDomain used to construct the UPN string for the authenticating user (optional).",
+			})
+		case userDnFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   userDnFlagName,
+				Target: &c.flagUserDn,
+				Usage:  "The base DN under which to perform user search (optional).",
+			})
+		case userAttrFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   userAttrFlagName,
+				Target: &c.flagUserAttr,
+				Usage:  "The attribute on user entry matching the username passed when authenticating (optional).",
+			})
+		case userFilterFlagName:
+			f.StringVar(&base.StringVar{
+				Name:   userFilterFlagName,
+				Target: &c.flagUserFilter,
+				Usage:  "A go template used to construct a LDAP user search filter (optional).",
```

Comment:
I think this is consistent with the design in ICU-080 but just want to double check that it was taken into account.

Reply:
Which RFC?

Reply:
Yes, it is.